lib/galaxy/tools/parameters/basic.py

```diff
@@ -2413,6 +2415,14 @@ def __init__(self, tool, input_source, context=None):
         input_source = ensure_input_source(input_source)
         SimpleTextToolParameter.__init__(self, tool, input_source)
 
+    def validate(self, value, trans=None):
+        super().validate(value, trans=trans)
+        file_source = trans.app.file_sources.get_file_source_path(value).file_source
+        user_context = ProvidesUserFileSourcesUserContext(trans)
+        user_has_access = file_source.user_has_access(user_context)
+        if not user_has_access:
+            raise ValueError(f"The user cannot access {value}.")
```

Comment:
Can you raise ParameterValueError here ?
```suggestion
            raise ParameterValueError(f"The user cannot access {value}.")
```
That should be handled and highlighted in the tool form then (bit tricky to test now that you can't select a file source like this, but you could set up the tool, restrict access, restart galaxy and hit execute).

Reply:
Sure! thanks for the tip!